app/views/unattended/provisioning_templates/snippet/saltstack_minion.erb

```diff
@@ -16,4 +19,12 @@ log_level: warning
 # * {'roles': ['webserver', 'frontend']}
 #
 %>
-grains: <%= host_param('salt_grains') ? host_param('salt_grains') : '{}' %>
+<% collected_grains = @host.derive_salt_grains(use_autosign: true) %>
+<% if collected_grains.empty? -%>
+grains: {}
+<% else -%>
+grains: <% collected_grains.each do |key, val| %>
+  "<%= "#{key}" -%>": "<%= "#{val}" -%>"
+  <% end -%>
+<% end -%>
```

Comment:
Can you test this with safemode turned on?

Reply:
Tested with safemode turned on.